beacon_chain/spec/state_transition.nim

```
@@ -572,8 +572,26 @@ proc makeBeaconBlock*(
              hash_tree_root(sync_aggregate),
              execution_payload_root.get,
              hash_tree_root(validator_changes.bls_to_execution_changes)])
-        elif consensusFork > ConsensusFork.Capella:
-          discard denebImplementationMissing
+        elif consensusFork == ConsensusFork.Deneb:
```

In these `when` branches covering forks, generally there's some default fallback even when there isn't, right now, any unknown fork (the codebase has no support for EIP6110), so it doesn't silently fall off the edge when the next one is added. This is because Nim doesn't have a `case` equivalent of runtime-`if` which checks for exhaustiveness.

https://github.com/status-im/nimbus-eth2/blob/1bc9f3a67ac519ab4f889ca19abfd74f5e07c205/beacon_chain/spec/forks.nim#L281-L303
and
https://github.com/status-im/nimbus-eth2/blob/1bc9f3a67ac519ab4f889ca19abfd74f5e07c205/beacon_chain/spec/forks.nim#L581-L594

show examples.